scipy/spatial/transform/_rotation.pyx

```diff
@@ -1934,13 +1946,92 @@ cdef class Rotation:
         cdef double[:] angles = _empty1(num_rotations)
 
         for ind in range(num_rotations):
-            angles[ind] = 2 * atan2(_norm3(quat[ind, :3]), abs(quat[ind, 3]))
+            angles[ind] = _quat_magnitude(quat[ind])
 
         if self._single:
             return angles[0]
         else:
             return np.asarray(angles)
 
+    @cython.embedsignature(True)
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def dist(Rotation self, Rotation other):
+        """Get the smallest angular distance(s) to another rotation.
+
+        For two rotations ``p`` and ``q``, the distance ``p.dist(q)`` is
+        equivalent to calculating ``(p * q.inv()).magnitude()``.
+
+        Parameters
+        ----------
+        other : `Rotation` instance
+            Object containing the rotations to measure against this one.
+
+        Returns
+        -------
+        dist : ndarray or float
+            Angle(s) in radians, float if object contains a single rotation
+            and ndarray if object contains multiple rotations. The angle
+            will always be in the range [0, pi].
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+        >>> p = R.from_quat([0, 0, 0, 1])
+        >>> q = R.from_quat(np.eye(4))
+        >>> p.dist(q)
+        array([3.14159265, 3.14159265, 3.14159265, 0.        ])
+
+        Distance to a single rotation:
+
+        >>> p.dist(q[0])
+        3.141592653589793
+        """
+        return (self * other.inv()).magnitude()
```

Comment:
Would it make sense to expose this as a rotation as well (as opposed to merely an angle)?

I have used a similar functionality from the `UnitQuaternion` type in the Rust library `nalgebra`.

The rotation from one quaternion to another is called [rotation_to()](https://docs.rs/nalgebra/latest/nalgebra/geometry/type.UnitQuaternion.html#method.rotation_to) and the corresponding angle is called [angle_to()](https://docs.rs/nalgebra/latest/nalgebra/geometry/type.UnitQuaternion.html#method.angle_to).

I have the feeling that this naming would be easier to understand than `dist()`, at least for me. Or is `dist` already an established term in some community?

BTW, shouldn't it be `other * self.inv()`? I know that this doesn't matter when taking the `magnitude()`, but for understanding the code it might be better? Or it could be something like `self.rotation_to(other).magnitude()`, which I think would be easier to grasp when reading the code. But it may have performance implications ...?

Reply:
I like these ideas! `dist` is a notional name - of the quaternion libraries I'm familiar with ([matlab](https://www.mathworks.com/help/aerotbx/flight-parameters-1.html?s_tid=CRUX_lftnav), [boost qvm](https://www.boost.org/doc/libs/1_80_0/libs/qvm/doc/html/index.html#_quaternion_operations), [numpy-quaternion](https://quaternion.readthedocs.io/en/latest/Package%20API%3A/quaternion/)), none of them have great ergonomics for dealing with rotations like this. That `nalgebra` library looks like good prior art, and I can update to match the name.

Reply:
Renamed `dist()` to `angle_to()` and added `rotation_to()` with tests.